pkg/controllermanager/controller/seed/seed_lifecycle_control.go

```diff
@@ -98,21 +131,29 @@ func (c *defaultControl) Reconcile(seedObj *gardencorev1beta1.Seed, key string)
 		seedLogger = logger.NewFieldLogger(logger.Logger, "seed", seed.Name)
 	)
 
-	// New seeds don't have conditions, i.e., gardenlet never reported anything yet. Let's wait until it sends the first heart beat.
+	seedLease := c.seedsLease.getDeepCopy(seed.Name)
+	defer func() {
+		c.seedsLease.set(seed.Name, seedLease)
+	}()
+
+	// New seeds don't have conditions - gardenlet never reported anything yet. Wait for grace period.
 	if len(seed.Status.Conditions) == 0 {
 		return true, nil
 	}
 
-	for _, condition := range seed.Status.Conditions {
-		// If the `GardenletReady` condition is not yet `Unknown` then check when it most recently sent a heartbeat and wait for the
-		// configured `monitorPeriod` before proceeding with any action.
-		if condition.Type == gardencorev1beta1.SeedGardenletReady && condition.Status != gardencorev1beta1.ConditionUnknown && !condition.LastUpdateTime.UTC().Before(time.Now().UTC().Add(-c.config.Controllers.Seed.MonitorPeriod.Duration)) {
```

Comment:
The object is being reconciled roughly on 10 seconds. If the lease was not updated during those 10 seconds we consider that the gardenlet condition is unknown.

Reply:
I've seen that - the question above was about the "why", and what is now happening with the monitor period?

Reply:
I am not using the MonitorPeriod, just checking if the stored lease is older than the observed. But i can change it if you would like to compare it to time.Now() as before.

Reply:
When we introduced the monitor period we discussed that it might be helpful to have this configurable (mainly for setups where the timeout may be prolonged, e.g., cross-continent seeds), so I'd prefer to keep this part as it was before.

Reply:
I see, yeah makes sense. I will change it then